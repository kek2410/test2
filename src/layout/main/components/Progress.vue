<template>
  <div style="min-height: 4px;">
    <v-progress-linear
      color="blue"
      :indeterminate="true"
      v-if="loading"
    ></v-progress-linear>
  </div>
</template>

<script lang="ts">
import { Vue, Component } from "vue-property-decorator";
import { Getter } from "vuex-class";

@Component({})
export default class Progress extends Vue {
  @Getter("loading") loading!: boolean;

  mounted() {
    // this.startBuffer();
  }
}
</script>
