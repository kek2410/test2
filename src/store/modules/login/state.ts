export declare interface State {
  userInfo: any;
  menuList?: any;
  isLogin: boolean;
  isLoginError: boolean;
  token: string;
  locale: string;
  themeColor: string;
}

const list = [
  {
    name: "DashBoard",
    title: "DashBoard"
  },
  {
    name: "TestList",
    title: "TestList"
  }
];

export const state: State = {
  userInfo: null,
  menuList: list,
  isLogin: false,
  isLoginError: false,
  token: "",
  locale: "ko",
  themeColor: "dark-grey"
};

export default state;
