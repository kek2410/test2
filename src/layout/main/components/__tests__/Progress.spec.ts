import { shallowMount, createLocalVue } from "@vue/test-utils";
import Progress from "../Progress.vue";
import Vuex from "vuex";
import Vuetify from "vuetify";
import state from "@/store/modules/login/state";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe("Progress.vue", (): void => {
  let store: any;
  let vuetify: any;

  beforeEach((): void => {
    vuetify = new Vuetify();
    store = new Vuex.Store({
      state: {
        loading: Boolean
      },
      mutations: {
        LOADING: (state: any, payload: boolean): any =>
          (state.loading = payload)
      },
      getters: {
        loading: (state: any): any => state.loading
      }
    });
  });

  it("Progress.vue > toMatchSnapshot", (): void => {
    const wr = shallowMount(Progress, {
      localVue,
      store,
      vuetify
    });
    expect(wr.html()).toMatchSnapshot();
  });

  it("Progress.vue > hide and show", async (): Promise<any> => {
    const wr = shallowMount(Progress, {
      localVue,
      store,
      vuetify
    });

    const findOne = wr.find("v-progress");

    expect(findOne.isVisible()).toBe(false);

    store.commit("LOADING", true);

    expect(findOne.isVisible()).toBe(true);
  });
});
