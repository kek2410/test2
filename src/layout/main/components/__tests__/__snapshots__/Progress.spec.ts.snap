// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress.vue Progress.vue > toMatchSnapshot 1`] = `
<div style="min-height: 4px;">
  <v-progress-linear-stub color="blue" value="0" active="true" buffervalue="100" height="4" indeterminate="true"></v-progress-linear-stub>
</div>
`;
